presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -293,54 +321,77 @@ public IPrestoSparkQueryExecution create(
         }
         else {
             checkArgument(sqlLocation.isPresent(), "sqlText or sqlLocation must be present");
-            sql = new String(metadataStorage.read(sqlLocation.get()), UTF_8);
+            byte[] sqlFileBytes = metadataStorage.read(sqlLocation.get());
+            if (sqlFileSizeInBytes.isPresent()) {
+                if (Integer.valueOf(sqlFileSizeInBytes.get()) != sqlFileBytes.length) {
+                    throw new PrestoException(
+                            MALFORMED_QUERY_FILE,
+                            format("sql file size %s is different from expected sqlFileSizeInBytes %s", sqlFileBytes.length, sqlFileSizeInBytes.get()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: sqlFileBytes.length. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 329. Consider formatting this data with &#39;%d&#39; instead.